amundsen_application/static/js/components/common/Table/index.tsx

```diff
@@ -182,6 +197,8 @@ const Table: React.FC<TableProps> = ({
   );
 
   if (data.length) {
+    checkValidData(data, fields);
```

Comment:
```suggestion
   if (!checkValidData(data, fields)) {
   throw new Error(INVALID_DATA_ERROR_MESSAGE);
 }
```

Reply:
what you get at the end is, that you can make ValidData type as interesting as you want so, if the type is validated you would have a type already casted and you will be able to do more things with it.